composer/core/time.py

```diff
@@ -298,7 +306,7 @@ def from_timestring(cls, timestring: str) -> Time:
         value = float(value)  # always parsing first as float b/c it could be scientific notation
         if unit != TimeUnit.DURATION:
             value = int(value)
-        return Time(value, unit)
+        return cls(value, unit)
```

Comment:
Similar to my other question - is this necessary to preserve the generic type annotation?

Reply:
Yeah, same thing here re: subclassing.